File: android/data/repositories/src/main/kotlin/com/gemwallet/android/data/repositories/gemstone/FileStore.kt
package com.gemwallet.android.data.repositories.gemstone

import com.gemwallet.android.data.service.store.LocalStore
import uniffi.gemstone.GemFileStore

class GemstoneFileStore(
    private val localStore: LocalStore,
) : GemFileStore {
    override fun saveFile(data: ByteArray, extension: String): String = localStore.save(data, extension)

    override fun saveNamedFile(data: ByteArray, fileName: String): String = localStore.saveNamed(data, fileName)

    override fun exists(fileName: String): Boolean = localStore.exists(fileName)

    override fun path(fileName: String): String = localStore.path(fileName)

    override fun remove(fileName: String) = localStore.remove(fileName)
}

File: ios/Packages/GemstoneServices/Sources/FileStore.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation
import protocol Gemstone.GemFileStore
import LocalStore

public final class GemstoneFileStore: GemFileStore, Sendable {
    private let store = LocalStore()

    public init() {}

    public func saveFile(data: Data, extension: String) throws -> String {
        try store.store(data, id: UUID().uuidString, documentType: `extension`)
    }

    public func saveNamedFile(data: Data, fileName: String) throws -> String {
        try store.store(data, fileName: fileName).path()
    }

    public func exists(fileName: String) -> Bool {
        store.exists(fileName)
    }

    public func path(fileName: String) -> String {
        store.url(for: fileName).path()
    }

    public func remove(fileName: String) throws {
        try store.remove(fileName)
    }
}
